illwill.nim

```diff
@@ -513,16 +711,45 @@ proc exitFullScreen() =
     eraseScreen()
     setCursorPos(0, 0)
 
-proc illwillInit*(fullScreen: bool = true) =
+when defined(posix):
+  proc enableMouse() =
+    stdout.write mouseTrackAny
+    stdout.flushFile()
+
+  proc disableMouse() =
+    stdout.write disableMouseTrackAny
+    stdout.flushFile()
+else:
+  proc enableMouse(hConsoleInput: Handle) =
+    var currentMode: DWORD
+    discard getConsoleMode(hConsoleInput, addr(currentMode))
+    discard setConsoleMode(hConsoleInput,
+      ENABLE_WINDOW_INPUT or ENABLE_MOUSE_INPUT or ENABLE_EXTENDED_FLAGS or
+      (currentMode and ENABLE_QUICK_EDIT_MODE.bitnot())
+    )
+
+  proc disableMouse(hConsoleInput: Handle, oldConsoleMode: DWORD) =
+    discard setConsoleMode(hConsoleInput,  oldConsoleMode) # TODO: REMOVE MOUSE OPTION ONLY?
+
+proc illwillInit*(fullScreen: bool = true, mouse: bool = false) =
   ## Initializes the terminal and enables non-blocking keyboard input. Needs
   ## to be called before doing anything with the library.
   ##
+  ## If `mouse = true` all mouse moves are tracked see `getMouse()`
```

Comment:
This doesn't read very well as an English sentence, please use better wording and punctuation.